src/object/cli_obj.c

```diff
@@ -2462,21 +2461,14 @@ obj_req_get_tgts(struct dc_object *obj, int *shard, daos_key_t *dkey,
 		flags = OBJ_TGT_FLAG_CLI_DISPATCH;
 		break;
 	case DAOS_OBJ_RPC_UPDATE:
-	case DAOS_OBJ_RPC_PUNCH:
 	case DAOS_OBJ_RPC_PUNCH_DKEYS:
 	case DAOS_OBJ_RPC_PUNCH_AKEYS:
-		if (opc == DAOS_OBJ_RPC_PUNCH) {
-			obj_ptr2shards(obj, &shard_idx, &shard_cnt, &grp_nr);
-		} else {
-			grp_nr = 1;
-			rc = obj_dkey2grpmemb(obj, dkey_hash, map_ver,
-					      &shard_idx, &shard_cnt);
-			if (rc != 0)
-				goto out;
-			D_DEBUG(DB_TRACE, "shard_idx %d shard_cnt %d\n",
-				(int)shard_idx, (int)shard_cnt);
-		}
-		flags = OBJ_TGT_FLAG_FW_LEADER_INFO;
```

Comment:
why remove the OBJ_TGT_FLAG_FW_LEADER_INFO flag? seems that is used by DTX leader selection logic

Reply:
oh, I moved this below in line 2511.  Since both PUNCH and SYNC will use the same way to locate the targets.

Reply:
well, for UPDATE/PUNCH_KEY also need to set the OBJ_TGT_FLAG_FW_LEADER_INFO right? your new code removed it for all

Reply:
ah, right.